packages/shadydom/src/utils.js

```diff
@@ -92,7 +92,14 @@ export const microtask = (callback) => {
   twiddle.textContent = content++;
 }
 
-export const hasDocumentContains = Boolean(document.contains);
+const hasDocumentContains = Boolean(document.contains);
+
+/** @type {function(!Document, !Node): boolean} */
+export const documentContains = (doc, node) => (
+  doc === node ||
+  (hasDocumentContains && doc[NATIVE_PREFIX + 'contains'](node)) ||
```

Comment:
Suggest doing this check once and writing different `documentContains` implementations based on the result.

Reply:
Yeah, that's good idea. Also, I just realized that I was trying to be too clever here with the short-circuiting: if `document.contains` and `document.documentElement` both exist but `doc` doesn't contain `node`, then it will end up running both `contains` calls.

Reply:
Added.